fix: set expire_after on applications

ludtkemorgan opened the pull request:
This PR addresses #5619 

- [x] Addresses the issue in full
- [ ] Addresses only certain aspects of the issue

## Description

When a listing is closed and `APPLICATION_DAYS_TILL_EXPIRY` environment variable is set then all applications attached to that listing should have the `expireAfter` value populated. Also if an application is submitted after listing is closed that application should also have the `expireAfter` value populated

## How Can This Be Tested/Reviewed?

1. Set `APPLICATION_DAYS_TILL_EXPIRY` environment variable to be a valid number
2. Reseed the staging values
3. Go to partners and open a listing that has applications attached 
4. Close the listing
5. Verify those applications should have a value on `expireAfter` that is the number you set in step 1 days in the future.
6. In the partner site submit an application to that closed listing
7. Verify that that application has a value on `expireAfter` that is the number you set in step 1 days after the `closed_at` date.

## Author Checklist:

- [ ] Added QA notes to the issue with applicable URLs
- [x] Reviewed in a desktop view
- [x] Reviewed in a mobile view
- [x] Reviewed considering accessibility
- [x] Added tests covering the changes
- [ ] Made corresponding changes to the documentation
- [ ] Ran `yarn generate:client` and/or created a migration when required

## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates

ludtkemorgan commented on api/test/unit/services/application.service.spec.ts:
```diff
@@ -812,1750 +812,2012 @@ describe('Testing application service', () => {
+      );
+    });
+
+    it('should create an application from partner site when listing is closed', async () => {
```
This is the new test added. The rest of this file is just to better organize the file by endpoint

emilyjablonski commented on api/src/services/listing.service.ts:
```diff
@@ -2665,6 +2669,26 @@ export class ListingService implements OnModuleInit {
+      this.logger.log(
+        `setting expireAfter of ${expireAfterDate.toDateString()} on ${
+          expiredApplications.count
+        } application for listing ${listingId}`,
```
**nitpick:** `application` --> `applications`

ludtkemorgan commented:
@emilyjablonski Sorry about not updating the label in time. This is ready to be reviewed. These are the main change since you reviewed
* The expire_after value is set on application is the listing is closed and this environment variable is set
* New test added for that functionality. Also refactored that test file to be organized by endpoint for easier readability
* Addressed your comments
